### PR TITLE
#9 GitHub actions build pipeline - attempt 1 of ?? (yay for YAML)

### LINKED ISSUE #33: Unit tests failing in main branch - need to get clean baseline.
**Describe the bug**
Unit tests for code in the main branch are failing.

**To Reproduce**
See CI Build - <https://github.com/SimonGeering/ChatterBot/pull/1/checks?check_run_id=963659950>

**Expected behavior**

### PR DESCRIPTION
**Pull request by SimonGeering**
Step 1 of #9 providing an initial CI/CD build pipeline; 

This is just the simple PR workflow that will compile the solution and run all the tests in the test project. 

Since we may want to filter tests by category later, so ill add annotations to the existing unit tests for that, as it will also give me something to PR into my own main branch to test the triggers.

This also fixes #33 with the resolution of a null reference bug.

### CODE REVIEW

**Comment by SimonGeering** [comment]
@benrick, after this PR is merged to `main` would you also be able to get the `dev` branch cut from `main` please? 
Not only would that allow the other trigger for this pipeline to be tested, but it would also be best if I were to make future PR branches off of `dev` as a step towards implementing the final structure in #23

File: .github/workflows/PullRequest.yml [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+on:
+  pull_request:
+    branches: 
+    - dev
+    - main
+
+jobs:
+  build:
+
+    runs-on: windows-latest #Can't use anything else as WPF build has a windows dependency - waiting on .NET MAUI
```
Comment (benrick) [inline .github/workflows/PullRequest.yml:12]: Yep, I'm hopeful that MAUI will have all the cool stuff we want.

**Review by benrick — APPROVED** [review]
Looks like a simple set of build steps.

**Comment by SimonGeering** [comment]
Obviously you will need to set-up the branch protection rules once it's in on your side. 
If I recall correctly the build has to run once before you can enforce it as a constraint on a PR, but you should still be able to enable `Require status checks to pass before merging`

**Pull request merged**